### DISCUSSION TITLE
[Bug]: SyntaxError: JSON.parse: unterminated string at line 1 column 12725 of the JSON data

### DISCUSSION THREAD

**Discussion by Oclair** [code4recovery/12-step-meeting-list#1370] [Q&A] [answered]
### Contact Details

feedback@aavienna.com

### Website With Issue

https://www.aavienna.com

### What happened?

Using TSML UI causes the following error message and prevents any front end functionality to work
SyntaxError: JSON.parse: unterminated string at line 1 column 12725 of the JSON data

Using Legacy UI restores complete functionality of the plugin

I have been experiencing this intermittent issue over several months through several plugin updates.
Sometimes I can make it work by activating and deactivating certain meetings in the list and then turning the TSML UI back on.

I am available to help fix this bug as required.
Thank you for your service!
Jason D in Vienna

**Comment by Oclair** [comment]
I am unsure if the following error messages from the log are relevant to this issue

`2024/02/20 07:44:35 [error] 55808#100330: *2299 FastCGI sent in stderr: "HP Warning:  Attempt to read property "after" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 45PHP message: PHP Warning:  Attempt to read property "before" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 39PHP message: PHP Warning:  Attempt to read property "link_before" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 42PHP message: PHP Warning:  Attempt to read property "link_after" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 43PHP message: PHP Warning:  Attempt to read property "after" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 45PHP message: PHP Warning:  Attempt to read property "before" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 39PHP message: PHP Warning:  Attempt to read property "link_before" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 42PHP message: PHP Warning:  Attempt to read property "link_after" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 43PHP message: PHP Warning:  Attempt to read property "after" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 45PHP message: PHP Warning:  Attempt to read property "before" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 39PHP message: PHP Warning:  Attempt to read property "link_before" on array in /usr/local/www/wordpressENAA/wp-content/themes/optimizer_pro/lib/includes/custom_walker.php on line 42PHP message: PHP Warning:  Attempt to read`

**Comment by joshreisner** [comment]
hi @Oclair do you have an example of this issue anywhere where we can see it?

if not, perhaps you could create a new, unlinked page on your site with the `[tsml_ui]` shortcode to see if it happens there?

TSML UI is reporting invalid JSON data, so we want to see what is happening to make it invalid - it could be those PHP warnings from your theme but we should confirm

**Reply by joshreisner** [reply]
thanks for setting up the page at https://www.aavienna.com/debugging/ - let us know when the error recurs and we can take a look!

or perhaps there was an update to optimizer pro and we won't see it again 🤞

**Reply by joshreisner** [reply]
thanks @Oclair  - looks like the issue has already corrected itself two hours after you posted that it wasn't working. the JSON file you posted is valid as well, unfortunately

next time it happens, try going straight to the json file in your browser's inspector > network tab and viewing or saving it from there

<img width="1728" alt="Screenshot 2024-03-10 at 8 11 23 AM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/e4ad7048-003d-46e7-b4ca-5b95e6c78233">

**Reply by joshreisner** [reply, marked as answer]
thanks @Oclair in the latest JSON you sent i can clearly see it ends prematurely, in the middle of the final meeting. 

bizarre that the same file works for me when i load it https://www.aavienna.com/wp-content/tsml-cache-18114522da.json?1710075206 - and the page works for me fine in firefox

i can see that the only data your version is missing is the final `,"last_contact":"2024-02-01","attendance_option":"online"}]`

i see from your screenshot that it is cached, i wonder if you reload it will it become complete?

also you mentioned that it only happens when you are logged in - if you try firefox in incognito mode would it still show the error?

**Reply by Oclair** [reply]
I did a force reload with the ctrl held down and voilla it works.  So this is a potential caching issue.  From past experience, I feel it is related to Firefox more than being logged in.  anyways thanks Josh, I hope my interactions are helpful!
 
G24
Jason D

**Comment by Oclair** [comment]
Hi Josh hopefully I will not be required to remove the meeting list functionality for too long.

>if not, perhaps you could create a new, unlinked page on your site with the [tsml_ui] shortcode to see if it happens there?

First of all let me try to shortcode
https://www.aavienna.com/debugging/
Works good!
![image](https://github.com/code4recovery/12-step-meeting-list/assets/619643/62a92042-049b-409d-9a03-efab950dd651)

Now lets see if the error returns turning TSML UI on the Setting page
ok so now the intermittent bug again has disappeared for utilizing the TSML UI appearance...

So at the moment it works, but this happens often when I add a new meeting for example.

>TSML UI is reporting invalid JSON data, so we want to see what is happening to make it invalid - it could be those PHP warnings from your theme but we should confirm

Please let me know what kind of log I could setup to report the intermittent bug, Maybe there is something in the source of the page when the error appears?

I have not seen any messages in the error log from nginx nor php.
Kind Regards, and thank you for your service!
Jason D

**Comment by Oclair** [comment]
This bug currently has resurfaced
![image](https://github.com/code4recovery/12-step-meeting-list/assets/619643/88d8f12e-bb67-4be5-86a3-3e4665a3a36e)

**Comment by Oclair** [comment]
> or perhaps there was an update to optimizer pro and we won't see it again 🤞

I am not sure the theme has the issue because it has to do with parsing the JSON meeting data within the plugin if I am not mistaken?

**Comment by Oclair** [comment]
for another reason I was in the wordpress/wp-content/ directory and found the following file
tsml-cache-18114522da.json
with the time stamp and following unix perms and ownership
-rw-r--r--   1 AAwww  AAwww  14022 Mar 10 13:53 **tsml-cache-18114522da.json

I have attached the file here
[tsml-cache-18114522da.json](https://github.com/code4recovery/12-step-meeting-list/files/14550570/tsml-cache-18114522da.json)

**Comment by Oclair** [comment]
I would suggest while basically its ok to use /wordpress/wp-content, a subdirectory for tsml cache data should be implemented and the cache data placed there.  Currently there are 7 similarly named .json files there.

![image](https://github.com/code4recovery/12-step-meeting-list/assets/619643/38f07bae-72d1-411b-a772-34a8f24b2231)

**Comment by Oclair** [comment]
> thanks @Oclair - looks like the issue has already corrected itself two hours after you posted that it wasn't working. the JSON file you posted is valid as well, unfortunately
> 
> next time it happens, try going straight to the json file in your browser's inspector > network tab and viewing or saving it from there
> <img alt="Screenshot 2024-03-10 at 8 11 23 AM" width="1728" src="https://private-user-images.githubusercontent.com/1551689/311520276-e4ad7048-003d-46e7-b4ca-5b95e6c78233.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAwODQ1NTgsIm5iZiI6MTcxMDA4NDI1OCwicGF0aCI6Ii8xNTUxNjg5LzMxMTUyMDI3Ni1lNGFkNzA0OC0wMDNkLTQ2ZTctYjRjYS01Yjk1ZTZjNzgyMzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTBUMTUyNDE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTYxMDRiMGQ5NzQyYTMxMjliMmRjMjhiZjJiZDJkN2ZjYzY4MDkzOWYxMmQ5YzI2YmMxY2NlZWFhMGNkODM2ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.7iFYfCWnARSv-l-cfbiODjA1Fcz_MITn8unsp_mcik4">


Hi thanks for the quick reply!

I am still getting the error on  https://www.aavienna.com/debugging/ which uses the [tsml_ui] shortcode

The plugin currently for the front end uses the classic UI so the error is not present at
https://www.aavienna.com/meetings/?tsml-day=any

**Reply by Oclair** [reply]
![image](https://github.com/code4recovery/12-step-meeting-list/assets/619643/f88e1fe4-ec41-488e-8702-3aecf9540920)

![image](https://github.com/code4recovery/12-step-meeting-list/assets/619643/abbea5f9-8661-4857-8459-f5436d4bab3c)

[SavedFromBrowser-tsml-cache-18114522da.json](https://github.com/code4recovery/12-step-meeting-list/files/14550758/SavedFromBrowser-tsml-cache-18114522da.json)

**Reply by Oclair** [reply]
the error seems to be related to Firefox for some reason as when I visit the page without being logged in it has no issue

Edge on win10
![image](https://github.com/code4recovery/12-step-meeting-list/assets/619643/203610a3-5887-4aac-b2a9-339eec1e81b9)
